main/src/cgeo/geocaching/storage/DataStore.java

```diff
@@ -1707,7 +1753,7 @@ public static Geocache loadCache(final String geocode, final EnumSet<LoadFlag> l
 
                 if (loadFlags.contains(LoadFlag.LOGS)) {
                     final Map<LogType, Integer> logCounts = loadLogCounts(cache.getGeocode());
-                    if (MapUtils.isNotEmpty(logCounts)) {
+                    if (logCounts != null && MapUtils.isNotEmpty(logCounts)) {
```

Comment:
If IDE bugs you about it, just add an assertion inside the "then" clause: `assert(logCounts != null);`. Also, if you're using IntelliJ IDEA, you can add a contract to `MapUtils.isNotEmpty` saying that `null` implies `false`, and you won't be bothered about it any longer.